src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs

```diff
@@ -154,6 +155,12 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
                     descriptors.Add(new ParameterDescriptor(ScriptConstants.SystemExecutionContextParameterName, typeof(ExecutionContext)));
                 }
 
+                if (!descriptors.Any(p => p.Type == typeof(CancellationToken)))
+                {
+                    // Add the CancellationToken. We need this to ensure we can cancel even if the Function hasn't requested it.
```

Comment:
I can't figure out how to make a link in mobile. But look at StartFunctionTimeout in core sdk. We won't start a timeout without a parameter. 

Also *we* use it. Just like ExecutionContext -- we need it even if the function doesn't ask for it. It's how we cancel the function even if they didn't ask for a timeout. 

I can explain better tomorrow if that still doesn't make sense.